2-collectors/scala-stream-collector/core/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala

```diff
@@ -401,5 +401,75 @@ class CollectorServiceSpec extends Specification {
           `Access-Control-Allow-Origin`(HttpOriginRange.`*`)
       }
     }
+
+    "cookieDomain" in {
+      "not return a domain" in {
+        "if a list of domains is not supplied in the config and there is no fallback domain" in {
+          val request = HttpRequest()
+          val cookieConfig = CookieConfig(true, "name", 5.seconds, None, None)
+          service.cookieDomain(request.headers, cookieConfig.domains, cookieConfig.fallbackDomain) shouldEqual None
+        }
+        "if a list of domains is supplied in the config but the Origin request header is empty and there is no fallback domain" in {
+          val request = HttpRequest()
+          val cookieConfig = CookieConfig(true, "name", 5.seconds, Some(List("domain.com")), None)
+          service.cookieDomain(request.headers, cookieConfig.domains, cookieConfig.fallbackDomain) shouldEqual None
+        }
+        "if none of the domains in the request's Origin header has a match in the list of domains supplied with the config and there is no fallback domain" in {
+          val origins = scala.collection.immutable.Seq(HttpOrigin("http", Host("origin.com")), HttpOrigin("http", Host("otherorigin.com", 8080)))
```

Comment:
:art: you can go with just `Seq` or `List` in tests that will make the code more readable I think